ili9341/spi_stm32f4.go

```diff
@@ -0,0 +1,142 @@
+// +build stm32f4
+
+package ili9341
+
+import (
+	"device/stm32"
+	"machine"
+)
+
+type spiDriver struct {
+	bus machine.SPI
+}
+
+func NewSpi(bus machine.SPI, dc, cs, rst machine.Pin) *Device {
```

Comment:
The Go convention is to keep abbreviations capitalized.

```suggestion
func NewSPI(bus machine.SPI, dc, cs, rst machine.Pin) *Device {
```

Reply:
This change will be inconsistent with the other ILI9341 SPI drivers (see my previous comment). Is that OK? Or should we update those as well?

Reply:
All of them have to be changed or you can't use code across multiple devices with ILI9341 as in some cases you'd have to call NewSPI and in others NewSpi. Note that changing it also breaks any existing code using ILI9341 package (do we care at the moment?).

Note that current ILI9341 drivers are limited as they are write-only which means you have a 76800 pixels buffer that you can't read even though you might want to. Keeping a copy of it in memory is 150KB (assuming 16bit color) which for most devices is most of your RAM so rarely an option. The ATSAMD51 write functions also seem to have some unexplained fixation on playing around with SPI CTRLB register while neither ATSAMD51 nor ILI9341 manual seem to imply any reason to do so. I plan to experiment a bit by:
- Changing the ILI9341 SPI writes for ATSAMD51 to check if my understanding of ATSAMD51 SPI and the effect on ILI9341 is correct
- Adding ILI9341 SPI reads for ATSAMD51 and adding some of the read commands to ILI9341

Reply:
My opinion is that `NewSPI` as per Ayke's suggestion is a better name more consistent with other drivers as well.  The other implementation for atsamd21 and atsamd51 should also be changed to keep things consistent, which will be a breaking change but I don't think that will be a big deal since the code is relatively newly released and the project is not at 1.0 yet.  The wioterminal examples will need to be updated to the new name if it is changed.  I'll go ahead and open a PR to update the names for the examples and the samd implementations.

Reply:
Updated the function names included in this PR (5f5f8a9), and the other drivers are being updated in PR #199 by @bgould. Marking this issue resolved.